contracts/rewards/src/contract/query.rs

```diff
@@ -35,9 +35,37 @@ pub fn rewards_pool(
     })
 }
 
+pub fn participation(
+    storage: &dyn Storage,
+    pool_id: PoolId,
+    epoch_num: Option<u64>,
+    block_height: u64,
+) -> Result<Option<msg::Participation>, ContractError> {
+    let epoch_num = match epoch_num {
+        Some(num) => num,
+        None => {
+            let current_params = state::load_params(storage);
+            Epoch::current(&current_params, block_height)?.epoch_num
+        }
+    };
+
+    let tally = state::load_epoch_tally(storage, pool_id, epoch_num)?;
+
+    match tally {
+        None => Ok(None),
+        Some(tally) => Ok(Some(msg::Participation {
```

Comment:
Looks good to me. I would still not recommend clients to compute actual rewards rate themselves from this response, since it's easy to introduce discrepancies between implementations (e.g. how thresholds are compared within the contract vs if floats are used in JS, missing error checks etc) and chances of client code becoming outdated.

Reply:
Returning the rewards by verifier map would be nice to have as well (or a separate query for it) but we can get by with the participation rates for now to measure performance. Maybe we introduce a separate REST API layer later that avoids adding more query code to the contract itself.